clients/drcachesim/tracer/tracer.cpp

```diff
@@ -856,6 +850,27 @@ create_buffer(per_thread_t *data)
     }
 }
 
+static size_t
+get_v2p_buffer_size()
+{
+    // The handoff interface requites we use dr_raw_mem_alloc and thus page alignment.
+    // We use one page which is 256 entries for 4K pages which is enough for
```

Comment:
This assumes 16-byte entries, but if the phys and virt addresses are more than 48 bits in size, then we'd need two markers to hold the value. Also, these markers come in pairs. So, worst case, the 4k page will hold 64 entries only, right?

Reply:
That is true: and since it's not signed, a high address will always need 2 entries.  That should only be vsyscall which is not really used, and all our arches are currently at 48-bit implementations, so I don't think we would actually see the worst case, but it is a future concern.  Updated the comment.

Reply:
Note that 64 is still plenty for the vast majority; some of the initial buffers need a lot (gzip needed 400 or so a couple of times) but in steady-state I would expect 64 to be more than enough.